File: DPBO-praktikum/1/Java/Tabel.java
public class Tabel {
    private int baris;
    private int kolom;

    Tabel(){

    }
    Tabel(int baris, int kolom){
        this.baris=baris;
        this.kolom=kolom;
    }
    public void set_baris(int baris){
        this.baris=baris;
    }
    public void set_kolom(int kolom){
        this.kolom=kolom;
    }
    public int get_baris(){
        return this.baris;
    }
    public int get_kolom(){
        return this.kolom;
    }
    public void buat_baris(String[] isi, int[] max, int garis, char type_line){
        int i=0,j=0;

        for(i=0; i<garis+13; i++){
            System.out.print(type_line);
        }
        System.out.println((""));
        for(i=0; i<isi.length; i++){
            System.out.print("| ");
            System.out.print(isi[i]+" ");
            for(j=0;j<max[i]-isi[i].length();j++){
                System.out.print(" ");
            }
        }
        System.out.println("|");
        for(i=0; i<garis+13; i++){
            System.out.print(type_line);
        }
        System.out.println((""));
    }
}

File: DPBO-praktikum/1/CPP/Tabel.cpp
//library yang dipakai
#include <iostream>
#include <string>
#include <list>
#include <array>
//standar library
using namespace std;

class Tabel
{
private:
    int baris;
    int kolom;
public:
//constructor
    Tabel(){

    }
    Tabel(int baris, int kolom){
        this->baris=baris;
        this->kolom=kolom;
    }
    void set_baris(int baris) {
        this->baris = baris; 
    }
    void set_kolom(int kolom) {
        this->kolom = kolom; 
    }
    int get_baris() { 
        return baris; 
    }
    int get_kolom() {
        return kolom; 
    }
    //menampilkan isi objek dalam bentuk sebaris tabel
    void buat_baris(array<string, 4> isi, int max[4], int line, char type_line) {
        for (int i = 0; i < line + 13; i++) {//menampilkan garis atas
            cout << type_line;
        }
        cout<<endl;
        cout << "| ";
        for (int i = 0; i < this->kolom; i++) {//menampilkan isi perkolom
            cout <<isi[i];
            for (int j = 0; j < max[i] - isi[i].length(); j++) {//menampilkan spasi
                cout << " ";
            }
            cout<<" | ";
        }
        cout << endl;
        for (int i = 0; i < line + 13; i++) {//menampilkan garis bawah
            cout << type_line;
        }
        cout << endl;
    }
    //destructor
    ~Tabel(){

    }
};

